plugin/src/init.server.lua

```diff
@@ -2,24 +2,65 @@
 local HttpService = game:GetService("HttpService")
 assert(plugin, "This code must run inside of a plugin")
 
+if game:GetService("RunService"):IsRunning() then
+	return
+end
+
 local toolbar = plugin:CreateToolbar("Luau") :: PluginToolbar
-local button = toolbar:CreateButton(
+
+local ConnectButton = toolbar:CreateButton(
 	"Luau Language Server Setup",
 	"Toggle Menu",
 	"rbxassetid://11115506617",
 	"Luau Language Server"
 ) :: PluginToolbarButton
 
-local widgetInfo = DockWidgetPluginGuiInfo.new(
-	-- widget info
-	Enum.InitialDockState.Float,
-	false,
-	false,
-	300,
-	200,
-	120,
-	70
-)
+local SettingsButton =
+	toolbar:CreateButton("Settings", "Open Settings", "rbxassetid://13997395868") :: PluginToolbarButton
+
+--#region Settings
+
+local AnalyticsService = game:GetService("AnalyticsService")
+
+local Settings = nil :: any
+
+local SettingsModule = AnalyticsService:FindFirstChild("LuauLSP_Settings")
+if not SettingsModule then
+	SettingsModule = script.DefaultSettings:Clone()
+	SettingsModule.Name = "LuauLSP_Settings"
+	SettingsModule.Parent = AnalyticsService
+end
+
+local function LoadSettings()
+	local result, parseError: any = loadstring(SettingsModule.Source)
+	if result == nil then
+		warn("[Luau Language Server] Could not load settings: " .. parseError)
+		Settings = nil
+		return
+	end
+	pcall(function()
+		Settings = result()
+	end)
+	if
+		type(Settings) ~= "table"
+		or type(Settings.port) ~= "number"
+		or type(Settings.startAutomatically) ~= "boolean"
+		or type(Settings.include) ~= "table"
+	then
+		Settings = nil
+		warn("[Luau Language Server] Could not load settings: invalid settings")
+	end
```

Comment:
Maybe this should provide better a more detailed message than just "invalid settings"? In particular, which setting is invalid